src/test/java/htsjdk/samtools/LongReadsTest.java

```diff
@@ -0,0 +1,227 @@
+                Assert.assertEquals(copyRecord, originalRecord);
+            }
+            Assert.assertFalse(copyIt.hasNext());
+        }
+    }
+
+    @Test
+    final public void testQueryOverlappingForEachRead() throws IOException {
+        final IOPath copiedBAMPath = createBAMCopyWithIndex(PAC_BIO_READS);
+
+        // for each mapped read in the original source BAM, query both the original and the copy for
+        // all reads that overlap that read's coordinates, and compare the results
+        try (final ReadsDecoder sourceDecoder =
+                     HtsDefaultRegistry.getReadsResolver().getReadsDecoder(ReadsBundle.resolveIndex(PAC_BIO_READS));
+             final ReadsDecoder originalDecoder =
+                     HtsDefaultRegistry.getReadsResolver().getReadsDecoder(ReadsBundle.resolveIndex(PAC_BIO_READS));
+             final ReadsDecoder copyDecoder =
+                     HtsDefaultRegistry.getReadsResolver().getReadsDecoder(ReadsBundle.resolveIndex(copiedBAMPath))) {
+            for (final SAMRecord samRecord : sourceDecoder) {
+                //skip unmapped records since we're going to use the read's coordinates as a query
+                if (!samRecord.getReadUnmappedFlag()) {
+                    try (final CloseableIterator<SAMRecord> originalIt = originalDecoder.queryOverlapping(
+                            samRecord.getContig(), samRecord.getAlignmentStart(), samRecord.getAlignmentEnd());
+                         final CloseableIterator<SAMRecord> copyIt = copyDecoder.queryOverlapping(
+                                 samRecord.getContig(), samRecord.getAlignmentStart(), samRecord.getAlignmentEnd())) {
+                        compareQueryResults(originalIt, copyIt);
+                    }
+                }
+            }
+        }
+    }
+
+    @Test
+    final public void testQueryContainedForEachRead() throws IOException {
+        final IOPath tmpBAMPath = createBAMCopyWithIndex(PAC_BIO_READS);
+
+        // for each mapped read in the original source BAM, query both the original and the copy for
+        // all reads that are contained within that read's coordinates, and compare the results
+        try (final ReadsDecoder sourceDecoder =
+                     HtsDefaultRegistry.getReadsResolver().getReadsDecoder(ReadsBundle.resolveIndex(PAC_BIO_READS));
+             final ReadsDecoder originalDecoder =
+                     HtsDefaultRegistry.getReadsResolver().getReadsDecoder(ReadsBundle.resolveIndex(PAC_BIO_READS));
+             final ReadsDecoder copyDecoder =
+                     HtsDefaultRegistry.getReadsResolver().getReadsDecoder(ReadsBundle.resolveIndex(tmpBAMPath))) {
+            for (final SAMRecord samRecord : sourceDecoder) {
+                //skip unmapped records since we're going to use the read's coordinates as a query
+                if (!samRecord.getReadUnmappedFlag()) {
+                    try (final CloseableIterator<SAMRecord> originalIt = originalDecoder.queryContained(
```

Comment:
We should probably add an overload of queryContained that takes a Locatable so you can just pass in the SamRecord.  Is it possible to make Locatable a subinterface of HtsInterval which returns smaller primitives?

Reply:
The Locatable overload might be a good idea. Though we'd probably want to add parallel ones for query and queryContained. I think you'd get incompatible return types if you tried to make Locatable to extend HtsInterval though.